packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -3053,3 +3119,135 @@ class _WhitespaceDirectionalityFormatter extends TextInputFormatter {
     return _ltrRegExp.hasMatch(String.fromCharCode(value)) ? TextDirection.ltr : TextDirection.rtl;
   }
 }
+
+class _ScribbleElement extends StatefulWidget {
+  const _ScribbleElement({Key? key, required this.child, required this.focusNode, required this.editableKey}): super(key: key);
+
+  final Widget child;
+  final FocusNode focusNode;
+  final GlobalKey editableKey;
+
+  @override
+  _ScribbleElementState createState() => _ScribbleElementState();
+}
+
+class _ScribbleElementState extends State<_ScribbleElement> implements ScribbleClient {
+  _ScribbleElementState(): _elementIdentifier = (_nextElementIdentifier++).toString();
+
+  @override
+  void initState() {
+    super.initState();
+    TextInput.registerScribbleElement(elementIdentifier, this);
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+    TextInput.unregisterScribbleElement(elementIdentifier);
+  }
+
+  RenderEditable? get renderEditable => widget.editableKey.currentContext?.findRenderObject() as RenderEditable?;
+
+  static int _nextElementIdentifier = 1;
+  final String _elementIdentifier;
+
+  @override
+  String get elementIdentifier => _elementIdentifier;
+
+  @override
+  void onScribbleFocus(Offset offset) {
+    widget.focusNode.requestFocus();
+    renderEditable?.selectPositionAt(from: offset, cause: SelectionChangedCause.keyboard);
+  }
+
+  @override
+  bool isInScribbleRect(Rect rect) {
+    final Rect _bounds = bounds;
+    if (renderEditable?.readOnly ?? false)
+      return false;
+    if (_bounds == Rect.zero)
+      return false;
+    if (!_bounds.overlaps(rect))
+      return false;
+    final Rect intersection = _bounds.intersect(rect);
+    return <Offset>[
+      intersection.topLeft,
+      intersection.topCenter,
+      intersection.topRight,
+      intersection.centerLeft,
+      intersection.center,
+      intersection.centerRight,
+      intersection.bottomLeft,
+      intersection.bottomCenter,
+      intersection.bottomRight
+    ].any((Offset point) {
+      final HitTestResult result = HitTestResult();
```

Comment:
Ah I see how difficult it would be to find hitable `RenderObject`s in a given `Rect`. Hitable `RenderBox`es can partially overlap each other. So never mind the previous comment about adding things to `RenderObject`. But still we probably shouldn't sample points. 

@goderbauer @justinmc any suggestions on how we can identify currently "interactable" render objects in a given rect?

Reply:
Yeah, sampling "random" points doesn't seem right here.

Also it seems we are doing this hit testing multiple times if there are multiple Scribbles on screen?

Can you describe what the hit test is trying to accomplish?

Reply:
@goderbauer there's an interesting scribble feature where you can scribble on screen without any text field focused. The text will be sent to the most relevant text field based on the scribble location. An example: in the "reminder" app you can launch the app and just scribble away, and a new "reminder" entry will be added.

To that end UIKit needs to ask us for "scribble elements" in a certain `Rect`. But it's tricky to tell if a `ScribbleClient` has a "hotspot" in the given a `Rect` (as it can be behind an `IgnorePointer` or obscured by another `ScribbleClient`, we would have to hit test every `Offset` in the `Rect`).

Reply:
Right, so the hit testing is last here, after making sure the `_ScribbleFocusable` intersects the rect that UIKit is interested in. I think in most cases that would just be a few elements. It also uses `any` so that in the nominal case where the whole text field is visible and interactive, only one point will need to be checked.

I am interested to hear some ideas on the best way to do this. FWIW I think this approach will work in most cases, since it is testing points on the intersection between the two rectangles. Unless someone has very large handwriting, that area should be relatively small.

We actually have a beta test going with a custom engine implementing this that seems to be going pretty well. Granted, that is on a separate branch as we haven't updated from flutter 1.22.2 – I will probably look into upgrading again once this makes it to the stable branch.

Reply:
What if we just hit test the center of the intersection and that's it?  Are there any real cases where that would fail and the user would reasonably expect it to succeed?

Unless I'm missing some case, it seems like hit testing the center of the intersection would only fail if the EditableText is more than half way obscured.  In reality, if the EditableText is obscured, the user probably would forgive us for not entering their text.  For example, if the field is partially (at least halfway) scrolled underneath the AppBar, it would be clear that the user needs to reveal it first to get it to work.

The main cases where this approach wouldn't work would be crazy polygon shaped widgets on top of text fields, which probably isn't vital to get correct in a feature like this.